In [14]:
import pandas as pd

A continuacion realizo la lectura de los csv

In [15]:
penetracion_p100hab = pd.read_csv('./DatasetsObligatorios/Penetración-poblacion.csv')
penetracion_pcia = pd.read_csv('./DatasetsObligatorios/Penetracion-hogares.csv')
accesos_baf = pd.read_csv('./DatasetsObligatorios/Totales Dial-BAf.csv')
accesos_baf_pcia = pd.read_csv('./DatasetsObligatorios/Dial-BAf.csv')
accesos_tec = pd.read_csv('./DatasetsObligatorios/Accesos Por Tecnología.csv')
accesos_vmd = pd.read_csv('./DatasetsObligatorios/Totales VMD.csv')
accesos_vel_prov = pd.read_csv('./DatasetsObligatorios/Accesos por velocidad.csv')
accesos_totales = pd.read_csv('./DatasetsObligatorios/Totales Accesos por velocidad.csv')
accesos_prov_vel = pd.read_csv('./DatasetsObligatorios/Velocidad_sin_Rangos.csv')
ingresos = pd.read_csv('./DatasetsObligatorios/Ingresos .csv')
accesos_loc = pd.read_csv('./DatasetsObligatorios/Acc_vel_loc_sinrangos.csv')
acceso_tec_loc = pd.read_csv('./DatasetsObligatorios/Accesos_tecnologia_localidad.csv')
mapa_conect = pd.read_csv('./DatasetsObligatorios/mapa_conectividad.csv')
metadata = pd.read_csv('./DatasetsComplementarios/Metadata_mundial.csv')
mundial = pd.read_csv('./DatasetsComplementarios/mundial.csv')


---

Comienzo a analizar los diferentes Datos de los csv para realizar un proceso de ETL, para su posterior EDA, y su final presentacion en un Dashboard.
* El objetivo de este analisis profundo es minimizar la cantidad de datasets y poseer los datos correctos y necesarios para su presentacion final en PowerBi.

In [16]:
penetracion_p100hab.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Año                       984 non-null    int64  
 1   Trimestre                 984 non-null    int64  
 2   Provincia                 984 non-null    object 
 3   Accesos por cada 100 hab  984 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 30.9+ KB


In [17]:
penetracion_pcia.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 4 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int64  
 1   Trimestre                     984 non-null    int64  
 2   Provincia                     984 non-null    object 
 3   Accesos por cada 100 hogares  984 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 30.9+ KB


In [18]:
#Combino columnas año, Trimestre, Provincia
df1_comb = pd.merge(penetracion_p100hab, penetracion_pcia, on =['Año', 'Trimestre','Provincia'])
df1_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 5 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int64  
 1   Trimestre                     984 non-null    int64  
 2   Provincia                     984 non-null    object 
 3   Accesos por cada 100 hab      984 non-null    float64
 4   Accesos por cada 100 hogares  984 non-null    float64
dtypes: float64(2), int64(2), object(1)
memory usage: 38.6+ KB


In [19]:
df1_comb.sample(2)

Año  Trimestre Provincia  Accesos por cada 100 hab  \
389  2020          1   Córdoba                 21.642327   
513  2018          4     Jujuy                 11.899484   

     Accesos por cada 100 hogares  
389                     70.688074  
513                     46.532266

---------------

In [20]:
accesos_tec.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 986 entries, 0 to 985
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           985 non-null    object 
 1   Trimestre     985 non-null    object 
 2   Provincia     984 non-null    object 
 3   ADSL          984 non-null    float64
 4   Cablemodem    984 non-null    float64
 5   Fibra óptica  984 non-null    float64
 6   Wireless      984 non-null    float64
 7   Otros         984 non-null    float64
 8   Total         984 non-null    float64
dtypes: float64(6), object(3)
memory usage: 69.5+ KB


In [21]:
accesos_tec.head(5)

Año Trimestre        Provincia      ADSL  Cablemodem  Fibra óptica  \
0  2024         1     Buenos Aires  203297.0   2790663.0     1799013.0   
1  2024         1  Capital Federal   70108.0   1177475.0      203756.0   
2  2024         1        Catamarca    5901.0     11025.0       56542.0   
3  2024         1            Chaco   11343.0     58929.0       64463.0   
4  2024         1           Chubut   26306.0     77217.0       26356.0   

   Wireless     Otros      Total  
0  150290.0  101294.0  5044557.0  
1    4493.0   48879.0  1504711.0  
2    1384.0      81.0    74933.0  
3    8404.0   12404.0   155543.0  
4   30867.0    9930.0   170676.0

In [22]:
accesos_tec.isna().sum()

Año             1
Trimestre       1
Provincia       2
ADSL            2
Cablemodem      2
Fibra óptica    2
Wireless        2
Otros           2
Total           2
dtype: int64

In [23]:
accesos_tec = accesos_tec.dropna()

In [24]:
#visualizo el Dataframe y encuentro '*' en Año y Trimestre. Reemplazo el *
accesos_tec = accesos_tec.copy()
accesos_tec['Año'].unique() 
accesos_tec['Año'] = accesos_tec['Año'].str.replace('[^0-9]', '').str.rstrip('*')
accesos_tec['Año'] = accesos_tec['Año'].astype(int)
#access_tec.head(5)
accesos_tec['Año'].unique() 

array([2024, 2023, 2022, 2021, 2020, 2019, 2018, 2017, 2016, 2015, 2014])

In [25]:
accesos_tec.info()

<class 'pandas.core.frame.DataFrame'>
Index: 984 entries, 0 to 983
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           984 non-null    int32  
 1   Trimestre     984 non-null    object 
 2   Provincia     984 non-null    object 
 3   ADSL          984 non-null    float64
 4   Cablemodem    984 non-null    float64
 5   Fibra óptica  984 non-null    float64
 6   Wireless      984 non-null    float64
 7   Otros         984 non-null    float64
 8   Total         984 non-null    float64
dtypes: float64(6), int32(1), object(2)
memory usage: 73.0+ KB


In [26]:
accesos_tec['Trimestre'] = accesos_tec['Trimestre'].str.replace('[^0-9]', '').str.rstrip('*')
accesos_tec['Trimestre'] = accesos_tec['Trimestre'].astype(int)

In [27]:
accesos_tec.info()

<class 'pandas.core.frame.DataFrame'>
Index: 984 entries, 0 to 983
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           984 non-null    int32  
 1   Trimestre     984 non-null    int32  
 2   Provincia     984 non-null    object 
 3   ADSL          984 non-null    float64
 4   Cablemodem    984 non-null    float64
 5   Fibra óptica  984 non-null    float64
 6   Wireless      984 non-null    float64
 7   Otros         984 non-null    float64
 8   Total         984 non-null    float64
dtypes: float64(6), int32(2), object(1)
memory usage: 69.2+ KB


In [37]:
# Combino el df previo creado con las mismas columnas
df2_comb = pd.merge(accesos_tec, df1_comb, on=['Año', 'Trimestre', 'Provincia'])
df2_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 11 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int32  
 1   Trimestre                     984 non-null    int32  
 2   Provincia                     984 non-null    object 
 3   ADSL                          984 non-null    float64
 4   Cablemodem                    984 non-null    float64
 5   Fibra óptica                  984 non-null    float64
 6   Wireless                      984 non-null    float64
 7   Otros                         984 non-null    float64
 8   Total                         984 non-null    float64
 9   Accesos por cada 100 hab      984 non-null    float64
 10  Accesos por cada 100 hogares  984 non-null    float64
dtypes: float64(8), int32(2), object(1)
memory usage: 77.0+ KB


In [45]:
df2_comb.sample(5)

Año  Trimestre            Provincia    ADSL  Cablemodem  Fibra óptica  \
949  2014          2             Misiones   56933       17594             4   
577  2018          1      Capital Federal  349995      968483         35535   
630  2017          3           Corrientes   56247       33993           947   
189  2022          2  Santiago Del Estero   16561       80338         11605   
975  2014          1            Río Negro   64886       24156           876   

     Wireless  Otros    Total  Accesos por cada 100 hab  \
949       150      8    74689                         6   
577      3164   9393  1366570                        44   
630      1334     27    92548                         8   
189      6618   2922   118044                        11   
975      3576   1572    95066                        13   

     Accesos por cada 100 hogares  
949                            23  
577                           106  
630                            32  
189                            47  
975                            44

In [53]:
# Paso todos los datos a interger
df2_comb = df2_comb.astype({
    'Provincia': 'string',
    'ADSL': 'int',
    'Cablemodem': 'int',
    'Wireless': 'int',
    'Otros': 'int',
    'Total': 'int',
    'Fibra óptica': 'int',
    'Accesos por cada 100 hab': 'int',
    'Accesos por cada 100 hogares': 'int'
})



In [55]:
df2_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 11 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   Año                           984 non-null    int32 
 1   Trimestre                     984 non-null    int32 
 2   Provincia                     984 non-null    string
 3   ADSL                          984 non-null    int32 
 4   Cablemodem                    984 non-null    int32 
 5   Fibra óptica                  984 non-null    int32 
 6   Wireless                      984 non-null    int32 
 7   Otros                         984 non-null    int32 
 8   Total                         984 non-null    int32 
 9   Accesos por cada 100 hab      984 non-null    int32 
 10  Accesos por cada 100 hogares  984 non-null    int32 
dtypes: int32(10), string(1)
memory usage: 46.3 KB


In [56]:
#Unifico DF con el mismo tipo de informacion.
df3_comb = pd.merge(df2_comb, accesos_baf_pcia, on=['Año', 'Trimestre', 'Provincia'])
df3_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 14 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int32  
 1   Trimestre                     984 non-null    int32  
 2   Provincia                     984 non-null    object 
 3   ADSL                          984 non-null    int32  
 4   Cablemodem                    984 non-null    int32  
 5   Fibra óptica                  984 non-null    int32  
 6   Wireless                      984 non-null    int32  
 7   Otros                         984 non-null    int32  
 8   Total_x                       984 non-null    int32  
 9   Accesos por cada 100 hab      984 non-null    int32  
 10  Accesos por cada 100 hogares  984 non-null    int32  
 11  Banda ancha fija              984 non-null    int64  
 12  Dial up                       982 non-null    float64
 13  Total

---

In [ ]:
accesos_totales.sample(3)

Año  Trimestre  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
28  2017          1           20653                   230159   
18  2019          3          241713                    80599   
29  2016          4           20724                   233398   

    Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  Entre 10 Mbps y 20 Mbps  \
28                4584534                  975027                  1129987   
18                3309137                 1026859                   806466   
29                4673339                  971094                  1038029   

    Entre 20 Mbps y 30 Mbps  Más de 30 Mbps   OTROS    Total  
28                   210668           47990       0  7199018  
18                  1141545         2314898  243467  9164684  
29                   169522           38590       0  7144696

In [ ]:
df4_comb = pd.merge(df3_comb, accesos_totales, on=['Año','Trimestre'])


In [ ]:
df4_comb.sample(3)

Año  Trimestre   Provincia   ADSL  Cablemodem  Fibra óptica  Wireless  \
566  2018          2     Neuquén  46633       52855           998     12649   
595  2018          1  Santa Cruz  18615        4474          1799      6734   
510  2018          4  Corrientes  55842       42418           796      1515   

     Otros  Total_x  Accesos por cada 100 hab  ...  Total_y  Hasta 512 kbps  \
566   2713   115848                        17  ...   115848            5972   
595   2783    34405                         9  ...    34405            7309   
510     30   100601                         9  ...   100601          138740   

     Entre 512 Kbps y 1 Mbps  Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  \
566                   219467                3316868                  979416   
595                   215406                3272724                  976956   
510                   201196                3337982                  882388   

     Entre 10 Mbps y 20 Mbps  Entre 20 Mbps y 30 Mbps  Más de 30 Mbps   OTROS  \
566                  1905839                  1000036          444391  238455   
595                  2068087                   862010          397977  236584   
510                  1472634                  1688748          599579  152388   

       Total  
566  8110444  
595  8037053  
510  8473655  

[3 rows x 23 columns]

In [ ]:
df_nac = pd.merge(accesos_baf, accesos_vmd, on=['Año', 'Trimestre'])
df_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 8 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Año                     41 non-null     int64  
 1   Trimestre               41 non-null     int64  
 2   Banda ancha fija        41 non-null     int64  
 3   Dial up                 41 non-null     int64  
 4   Total                   41 non-null     int64  
 5   Periodo                 41 non-null     object 
 6   Mbps (Media de bajada)  41 non-null     float64
 7   Trimestre.1             41 non-null     object 
dtypes: float64(1), int64(5), object(2)
memory usage: 2.7+ KB


In [ ]:
df2_nac = pd.merge(df_nac, accesos_vel_prov, on =['Año', 'Trimestre'])
df3_nac = pd.merge(df2_nac, ingresos, on =['Año', 'Trimestre'])
df4_nac = pd.merge(df3_nac, accesos_tec, on =['Año', 'Trimestre'])

In [ ]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Total_x                    22464 non-null  int64  
 5   Periodo_x                  22464 non-null  object 
 6   Mbps (Media de bajada)     22464 non-null  float64
 7   Trimestre.1                22464 non-null  object 
 8   Provincia_x                22464 non-null  object 
 9   HASTA 512 kbps             22464 non-null  float64
 10  + 512 Kbps - 1 Mbps        22464 non-null  float64
 11  + 1 Mbps - 6 Mbps          22464 non-null  float64
 12  + 6 Mbps - 10 Mbps         22464 non-null  float64
 13  + 10 Mbps - 20 Mbps        22464 non-null  flo

In [ ]:
df_partidos = pd.merge(acceso_tec_loc, mapa_conect, on = ['Provincia', 'Partido', 'Localidad'])
df_partidos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2826 entries, 0 to 2825
Data columns (total 26 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Provincia       2826 non-null   object 
 1   Partido         2826 non-null   object 
 2   Localidad       2826 non-null   object 
 3   Link Indec      2826 non-null   object 
 4   ADSL_x          2826 non-null   int64  
 5   CABLEMODEM      2826 non-null   int64  
 6   DIAL UP         2826 non-null   int64  
 7   FIBRA OPTICA    2826 non-null   int64  
 8   OTROS           2826 non-null   int64  
 9   SATELITAL       2826 non-null   int64  
 10  WIMAX           2826 non-null   int64  
 11  WIRELESS        2826 non-null   int64  
 12  Total general   2826 non-null   int64  
 13  Población       2826 non-null   int64  
 14  ADSL_y          2826 non-null   object 
 15  Cablemódem      2826 non-null   object 
 16  Dial Up         2826 non-null   object 
 17  Fibra óptica    2826 non-null   o

In [ ]:
df2_partidos = pd.merge(df_partidos, accesos_loc, on = ['Provincia', 'Partido', 'Localidad'])
df2_partidos.head(5)

Provincia     Partido               Localidad Link Indec_x  ADSL_x  \
0  BUENOS AIRES  25 de Mayo              25 de Mayo      6854100     755   
1  BUENOS AIRES  25 de Mayo               Del Valle      6854020     181   
2  BUENOS AIRES  25 de Mayo       Gobernador Ugarte      6854040       0   
3  BUENOS AIRES  25 de Mayo  Norberto de la Riestra      6854060       0   
4  BUENOS AIRES  25 de Mayo        Lucas Monteverde      6854050       0   

   CABLEMODEM  DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  ...  680 Mbps  \
0        4600        0             2      0        742  ...       NaN   
1           0        0             0     10          1  ...       NaN   
2           0        0             0      0          0  ...       NaN   
3         782        0           496     16          0  ...       NaN   
4           0        0             0      0          0  ...       NaN   

   700 Mbps  800 Mbps  850 Mbps 900 Mbps 999 Mbps 1000 Mbps 1024 Mbps  \
0       NaN       NaN       NaN      NaN      NaN       NaN       NaN   
1       NaN       NaN       NaN      NaN      NaN       NaN       NaN   
2       NaN       NaN       NaN      NaN      NaN       NaN       NaN   
3       NaN       NaN       NaN      NaN      NaN       NaN       NaN   
4       NaN       NaN       NaN      NaN      NaN       NaN       NaN   

  4000 Mbps 10000 Mbps  
0       NaN        NaN  
1       NaN        NaN  
2       NaN        NaN  
3       NaN        NaN  
4       NaN        NaN  

[5 rows x 148 columns]

Para finalizar, me encuentro con 4 DF con la informacion que necesito, los cuales son:
df4_comb, df4_nac, df2_partidos, indic_me

---

In [ ]:
# Evaluo los resultados del trabajo
igual_1 = (df4_comb['Total'] == df4_comb['Total_x']).sum()
igual_2 = (df4_comb['Total'] == df4_comb['Total_y']).sum()
igual_3 = (df4_comb['Total_x'] == df4_comb['Total_y']).sum()

print(igual_1)
print(igual_2)
print(igual_3)

0
0
984


Visualizo que no existen coincidencias entre 'Total' y 'Total_x', tampoco entre 'Total' y 'Total_y'; pero si hay 984 coincidencias entre 'Total_x' y 'Total_y', se elimina la columna.


In [ ]:
df4_comb = df4_comb.drop('Total_y', axis=1)
df4_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 22 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int32  
 1   Trimestre                     984 non-null    int32  
 2   Provincia                     984 non-null    object 
 3   ADSL                          984 non-null    int32  
 4   Cablemodem                    984 non-null    int32  
 5   Fibra óptica                  984 non-null    int32  
 6   Wireless                      984 non-null    int32  
 7   Otros                         984 non-null    int32  
 8   Total_x                       984 non-null    int32  
 9   Accesos por cada 100 hab      984 non-null    int32  
 10  Accesos por cada 100 hogares  984 non-null    float64
 11  Banda ancha fija              984 non-null    int64  
 12  Dial up                       982 non-null    float64
 13  Hasta

In [ ]:
# Renombro algunas columnas necesarias. 
df4_comb = df4_comb.rename(columns = {'Total': 'Total suma Mbps'})
df4_comb = df4_comb.rename(columns = {'Total_x': 'Total conexion'})
df4_comb = df4_comb.rename(columns = {'OTROS': 'Otros Mbps'})
df4_comb = df4_comb.rename(columns = {'HASTA 512 kbps': 'Hasta 512 kbps'})

In [ ]:
df4_comb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 22 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           984 non-null    int32  
 1   Trimestre                     984 non-null    int32  
 2   Provincia                     984 non-null    object 
 3   ADSL                          984 non-null    int32  
 4   Cablemodem                    984 non-null    int32  
 5   Fibra óptica                  984 non-null    int32  
 6   Wireless                      984 non-null    int32  
 7   Otros                         984 non-null    int32  
 8   Total conexion                984 non-null    int32  
 9   Accesos por cada 100 hab      984 non-null    int32  
 10  Accesos por cada 100 hogares  984 non-null    float64
 11  Banda ancha fija              984 non-null    int64  
 12  Dial up                       982 non-null    float64
 13  Hasta

In [ ]:
df_group = df4_comb.groupby(['Año', 'Trimestre', 'Provincia']).sum()
df_group.sample(5)

ADSL  Cablemodem  Fibra óptica  Wireless  Otros  \
Año  Trimestre Provincia                                                       
2014 4         Santa Cruz   17553         448           388      4395   2929   
     2         La Pampa     15957       32600           513      5869    689   
2015 4         San Luis     11432        1025           354        30    377   
2017 3         Santa Fe    322953      290655          6338      4897  37499   
2023 4         Misiones     25840       79000         75382     12928   7183   

                           Total conexion  Accesos por cada 100 hab  \
Año  Trimestre Provincia                                              
2014 4         Santa Cruz           25713                         8   
     2         La Pampa             55628                        16   
2015 4         San Luis             13218                         2   
2017 3         Santa Fe            662342                        19   
2023 4         Misiones            200333                        15   

                           Accesos por cada 100 hogares  Banda ancha fija  \
Año  Trimestre Provincia                                                    
2014 4         Santa Cruz                     27.802679             25277   
     2         La Pampa                       49.707596             55048   
2015 4         San Luis                        9.705769             12901   
2017 3         Santa Fe                       60.906356            662086   
2023 4         Misiones                       56.270000            200333   

                           Dial up  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
Año  Trimestre Provincia                                                      
2014 4         Santa Cruz    436.0           44075                   554749   
     2         La Pampa      580.0           41064                   656408   
2015 4         San Luis      317.0           34243                   427394   
2017 3         Santa Fe      256.0           19022                   217997   
2023 4         Misiones        0.0           29708                    71742   

                           Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  \
Año  Trimestre Provincia                                                   
2014 4         Santa Cruz                5084556                  496251   
     2         La Pampa                  5149574                  341689   
2015 4         San Luis                  5049640                  726740   
2017 3         Santa Fe                  4062226                 1062810   
2023 4         Misiones                   900253                  978108   

                           Entre 10 Mbps y 20 Mbps  Entre 20 Mbps y 30 Mbps  \
Año  Trimestre Provincia                                                      
2014 4         Santa Cruz                   276254                     4371   
     2         La Pampa                     147273                      478   
2015 4         San Luis                     639011                    17568   
2017 3         Santa Fe                    1592304                   353480   
2023 4         Misiones                     697232                   350290   

                           Más de 30 Mbps  Otros Mbps  Total suma Mbps  
Año  Trimestre Provincia                                                
2014 4         Santa Cruz           15229           0          6475485  
     2         La Pampa             12259           0          6348745  
2015 4         San Luis             22170           0          6916766  
2017 3         Santa Fe            245790           0          7553629  
2023 4         Misiones           8224736      295592         11547661

-----

In [ ]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 27 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Total_x                    22464 non-null  int64  
 5   Periodo_x                  22464 non-null  object 
 6   Mbps (Media de bajada)     22464 non-null  float64
 7   Trimestre.1                22464 non-null  object 
 8   Provincia_x                22464 non-null  object 
 9   HASTA 512 kbps             22464 non-null  float64
 10  + 512 Kbps - 1 Mbps        22464 non-null  float64
 11  + 1 Mbps - 6 Mbps          22464 non-null  float64
 12  + 6 Mbps - 10 Mbps         22464 non-null  float64
 13  + 10 Mbps - 20 Mbps        22464 non-null  flo

In [ ]:
df4_nac = df4_nac.rename(columns = {'HASTA 512 kbps': 'Hasta 512 kbps'})

In [ ]:
df4_nac = df4_nac.drop(['Periodo_y','Periodo_x'], axis = 1)
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 25 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Total_x                    22464 non-null  int64  
 5   Mbps (Media de bajada)     22464 non-null  float64
 6   Trimestre.1                22464 non-null  object 
 7   Provincia_x                22464 non-null  object 
 8   Hasta 512 kbps             22464 non-null  float64
 9   + 512 Kbps - 1 Mbps        22464 non-null  float64
 10  + 1 Mbps - 6 Mbps          22464 non-null  float64
 11  + 6 Mbps - 10 Mbps         22464 non-null  float64
 12  + 10 Mbps - 20 Mbps        22464 non-null  float64
 13  + 20 Mbps - 30 Mbps        22464 non-null  flo

In [ ]:
df4_nac

Año  Trimestre  Banda ancha fija  Dial up   Total_x  \
0      2024          1          11621363    11793  11633156   
1      2024          1          11621363    11793  11633156   
2      2024          1          11621363    11793  11633156   
3      2024          1          11621363    11793  11633156   
4      2024          1          11621363    11793  11633156   
...     ...        ...               ...      ...       ...   
22459  2014          1           6362108    36290   6398398   
22460  2014          1           6362108    36290   6398398   
22461  2014          1           6362108    36290   6398398   
22462  2014          1           6362108    36290   6398398   
22463  2014          1           6362108    36290   6398398   

       Mbps (Media de bajada)   Trimestre.1   Provincia_x  Hasta 512 kbps  \
0                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
1                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
2                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
3                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
4                  139.150000  Ene-Mar 2024  Buenos Aires         26002.0   
...                       ...           ...           ...             ...   
22459                3.617127  Ene-Mar 2014       Tucumán             6.0   
22460                3.617127  Ene-Mar 2014       Tucumán             6.0   
22461                3.617127  Ene-Mar 2014       Tucumán             6.0   
22462                3.617127  Ene-Mar 2014       Tucumán             6.0   
22463                3.617127  Ene-Mar 2014       Tucumán             6.0   

       + 512 Kbps - 1 Mbps  ...     OTROS    Total_y  \
0                  22510.0  ...  125147.0  5044557.0   
1                  22510.0  ...  125147.0  5044557.0   
2                  22510.0  ...  125147.0  5044557.0   
3                  22510.0  ...  125147.0  5044557.0   
4                  22510.0  ...  125147.0  5044557.0   
...                    ...  ...       ...        ...   
22459              34672.0  ...       0.0   130032.0   
22460              34672.0  ...       0.0   130032.0   
22461              34672.0  ...       0.0   130032.0   
22462              34672.0  ...       0.0   130032.0   
22463              34672.0  ...       0.0   130032.0   

       Ingresos (miles de pesos)          Provincia_y      ADSL  Cablemodem  \
0                   2.804191e+08         Buenos Aires  203297.0   2790663.0   
1                   2.804191e+08      Capital Federal   70108.0   1177475.0   
2                   2.804191e+08            Catamarca    5901.0     11025.0   
3                   2.804191e+08                Chaco   11343.0     58929.0   
4                   2.804191e+08               Chubut   26306.0     77217.0   
...                          ...                  ...       ...         ...   
22459               2.984054e+06           Santa Cruz   18927.0       100.0   
22460               2.984054e+06             Santa Fe  322662.0    174296.0   
22461               2.984054e+06  Santiago Del Estero   32567.0      3598.0   
22462               2.984054e+06     Tierra Del Fuego   21618.0      2837.0   
22463               2.984054e+06              Tucumán  129717.0        83.0   

       Fibra óptica  Wireless     Otros      Total  
0         1799013.0  150290.0  101294.0  5044557.0  
1          203756.0    4493.0   48879.0  1504711.0  
2           56542.0    1384.0      81.0    74933.0  
3           64463.0    8404.0   12404.0   155543.0  
4           26356.0   30867.0    9930.0   170676.0  
...             ...       ...       ...        ...  
22459         381.0    4092.0    3264.0    26764.0  
22460        3059.0    5951.0     644.0   506612.0  
22461          19.0     915.0      34.0    37133.0  
22462         648.0       1.0    2934.0    28038.0  
22463         121.0      13.0      98.0   130032.0  

[22464 rows x 25 columns]

In [ ]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 25 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Total_x                    22464 non-null  int64  
 5   Mbps (Media de bajada)     22464 non-null  float64
 6   Trimestre.1                22464 non-null  object 
 7   Provincia_x                22464 non-null  object 
 8   Hasta 512 kbps             22464 non-null  float64
 9   + 512 Kbps - 1 Mbps        22464 non-null  float64
 10  + 1 Mbps - 6 Mbps          22464 non-null  float64
 11  + 6 Mbps - 10 Mbps         22464 non-null  float64
 12  + 10 Mbps - 20 Mbps        22464 non-null  float64
 13  + 20 Mbps - 30 Mbps        22464 non-null  flo

In [ ]:
#Revision de Iguales
igual_4 = (df4_nac['Total_x'] == df4_nac['Total_y']).sum()
print(igual_4)

0


In [ ]:
df4_nac.sample(5)

Año  Trimestre  Banda ancha fija  Dial up  Total_x  \
17136  2016          2           7097604    32475  7130079   
17374  2016          1           6874704    32652  6907356   
9344   2019          3           9142891    21793  9164684   
14022  2017          3           7696594    28349  7724943   
16911  2016          2           7097604    32475  7130079   

       Mbps (Media de bajada)    Trimestre.1 Provincia_x  Hasta 512 kbps  \
17136                5.423531   Abr-Jun 2016    San Luis            12.0   
17374                5.083702   Ene-Mar 2016       Chaco           194.0   
9344                24.180000  Jul-Sept 2019     Córdoba            83.0   
14022                8.339048  Jul-Sept 2017     Formosa          1063.0   
16911                5.423531   Abr-Jun 2016     Formosa          1675.0   

       + 512 Kbps - 1 Mbps  ...   OTROS   Total_y  Ingresos (miles de pesos)  \
17136                  4.0  ...     0.0   13950.0               6.534241e+06   
17374               8755.0  ...     0.0   86650.0               5.936845e+06   
9344                5824.0  ...  7353.0  782341.0               2.165275e+07   
14022               3716.0  ...     0.0   31132.0               1.006600e+07   
16911               5401.0  ...     0.0   28836.0               6.534241e+06   

            Provincia_y       ADSL  Cablemodem  Fibra óptica  Wireless  \
17136      Buenos Aires  1579448.0   1305949.0      113070.0   21163.0   
17374  Tierra Del Fuego    22636.0      4834.0         665.0       1.0   
9344            Formosa    21777.0      8359.0        1476.0   14292.0   
14022        Corrientes    56247.0     33993.0         947.0    1334.0   
16911         Río Negro    67031.0     28100.0         939.0    3300.0   

         Otros      Total  
17136  25661.0  3045291.0  
17374   2350.0    30486.0  
9344     162.0    46066.0  
14022     27.0    92548.0  
16911   1334.0   100704.0  

[5 rows x 25 columns]

In [ ]:
#Elimino los '.'
df4_nac['Hasta 512 kbps'] = df4_nac['Hasta 512 kbps'].astype(str).str.replace('.',',')

In [ ]:
# Creo una Funcion para reemplazar los '+' y los '-'
def renombre_col(columna):
    columna = columna.replace('+', 'Entre')
    columna = columna.replace('-','y')
    return columna

In [ ]:
df4_nac.rename(columns=renombre_col, inplace=True)

In [ ]:
df4_nac.sample(5)

Año  Trimestre  Banda ancha fija  Dial up   Total_x  \
142    2024          1          11621363    11793  11633156   
12560  2018          2           8083533    26911   8110444   
3595   2022          1          10611390    12619  10624009   
12872  2018          1           8009981    27072   8037053   
859    2023          3          11395115    11793  11406908   

       Mbps (Media de bajada)    Trimestre.1 Provincia_x Hasta 512 kbps  \
142                 139.15000   Ene-Mar 2024     Córdoba           99,0   
12560                13.84954   Abr-Jun 2018  Santa Cruz           26,0   
3595                 55.11000   Ene-Mar 2022     Córdoba          100,0   
12872                13.22134   Ene-Mar 2018     Formosa         1110,0   
859                 129.67000  Jul-Sept 2023    La Rioja           18,0   

       Entre 512 Kbps y 1 Mbps  ...    OTROS    Total_y  \
142                     8056.0  ...  14710.0  1082199.0   
12560                    120.0  ...   4267.0    33772.0   
3595                   12782.0  ...  13104.0  1003803.0   
12872                   3803.0  ...      0.0    32320.0   
859                      459.0  ...      0.0    87701.0   

       Ingresos (miles de pesos)       Provincia_y     ADSL  Cablemodem  \
142                 2.804191e+08  Tierra Del Fuego   9357.0     48005.0   
12560               1.431947e+07           Formosa  23543.0      7311.0   
3595                5.521931e+07        Santa Cruz  10537.0     23420.0   
12872               1.317146e+07           Formosa  23556.0      7095.0   
859                 1.331066e+08        Santa Cruz   8376.0     23931.0   

       Fibra óptica  Wireless  Otros    Total  
142          3684.0     177.0  332.0  61555.0  
12560          18.0   13625.0   57.0  44554.0  
3595        10766.0    3974.0  405.0  49102.0  
12872          17.0    1603.0   57.0  32328.0  
859         11270.0    3660.0  750.0  47987.0  

[5 rows x 25 columns]

In [ ]:
# Redondeo los numeros 
df4_nac = df4_nac.round(2)

In [ ]:
df4_nac.sample(5)

Año  Trimestre  Banda ancha fija  Dial up   Total_x  \
22455  2014          1           6362108    36290   6398398   
3222   2022          2          10946248    12436  10958684   
5293   2021          2           9852702    10382   9863084   
20616  2014          4           6559172    39324   6598496   
17515  2016          1           6874704    32652   6907356   

       Mbps (Media de bajada)   Trimestre.1 Provincia_x Hasta 512 kbps  \
22455                    3.62  Ene-Mar 2014     Tucumán            6,0   
3222                    58.44  Abr-Jun 2022     Neuquén          134,0   
5293                    45.63  Abr-Jun 2021      Chubut          104,0   
20616                    4.16  Oct-Dic 2014  Santa Cruz          156,0   
17515                    5.08  Ene-Mar 2016       Jujuy         2543,0   

       Entre 512 Kbps y 1 Mbps  ...    OTROS   Total_y  \
22455                  34672.0  ...      0.0  130032.0   
3222                     999.0  ...  18717.0  161968.0   
5293                     105.0  ...   5172.0  139237.0   
20616                   1400.0  ...      0.0   25703.0   
17515                   8558.0  ...      0.0   40295.0   

       Ingresos (miles de pesos)   Provincia_y       ADSL  Cablemodem  \
22455                 2984054.21     Río Negro    64886.0     24156.0   
3222                 60419500.62    Corrientes    38468.0     75769.0   
5293                 38239666.63      Misiones    44960.0     87861.0   
20616                 3950440.78  Buenos Aires  1575978.0   1070135.0   
17515                 5936844.89    Santa Cruz    15654.0      2723.0   

       Fibra óptica  Wireless    Otros      Total  
22455         876.0    3576.0   1572.0    95066.0  
3222        18181.0    7256.0   5323.0   144997.0  
5293        10376.0    5317.0   7662.0   156176.0  
20616      120528.0   17728.0  32038.0  2816407.0  
17515         396.0    5088.0   2426.0    26287.0  

[5 rows x 25 columns]

In [ ]:
#Elimino las columnas redundantes
df4_nac = df4_nac.drop(['Total_y','Total_x'], axis = 1)
df4_nac

Año  Trimestre  Banda ancha fija  Dial up  Mbps (Media de bajada)  \
0      2024          1          11621363    11793                  139.15   
1      2024          1          11621363    11793                  139.15   
2      2024          1          11621363    11793                  139.15   
3      2024          1          11621363    11793                  139.15   
4      2024          1          11621363    11793                  139.15   
...     ...        ...               ...      ...                     ...   
22459  2014          1           6362108    36290                    3.62   
22460  2014          1           6362108    36290                    3.62   
22461  2014          1           6362108    36290                    3.62   
22462  2014          1           6362108    36290                    3.62   
22463  2014          1           6362108    36290                    3.62   

        Trimestre.1   Provincia_x Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
0      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
1      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
2      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
3      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
4      Ene-Mar 2024  Buenos Aires        26002,0                  22510.0   
...             ...           ...            ...                      ...   
22459  Ene-Mar 2014       Tucumán            6,0                  34672.0   
22460  Ene-Mar 2014       Tucumán            6,0                  34672.0   
22461  Ene-Mar 2014       Tucumán            6,0                  34672.0   
22462  Ene-Mar 2014       Tucumán            6,0                  34672.0   
22463  Ene-Mar 2014       Tucumán            6,0                  34672.0   

       Entre 1 Mbps y 6 Mbps  ...  Entre 30 Mbps     OTROS  \
0                   221185.0  ...      4068292.0  125147.0   
1                   221185.0  ...      4068292.0  125147.0   
2                   221185.0  ...      4068292.0  125147.0   
3                   221185.0  ...      4068292.0  125147.0   
4                   221185.0  ...      4068292.0  125147.0   
...                      ...  ...            ...       ...   
22459                83210.0  ...            0.0       0.0   
22460                83210.0  ...            0.0       0.0   
22461                83210.0  ...            0.0       0.0   
22462                83210.0  ...            0.0       0.0   
22463                83210.0  ...            0.0       0.0   

       Ingresos (miles de pesos)          Provincia_y      ADSL  Cablemodem  \
0                   2.804191e+08         Buenos Aires  203297.0   2790663.0   
1                   2.804191e+08      Capital Federal   70108.0   1177475.0   
2                   2.804191e+08            Catamarca    5901.0     11025.0   
3                   2.804191e+08                Chaco   11343.0     58929.0   
4                   2.804191e+08               Chubut   26306.0     77217.0   
...                          ...                  ...       ...         ...   
22459               2.984054e+06           Santa Cruz   18927.0       100.0   
22460               2.984054e+06             Santa Fe  322662.0    174296.0   
22461               2.984054e+06  Santiago Del Estero   32567.0      3598.0   
22462               2.984054e+06     Tierra Del Fuego   21618.0      2837.0   
22463               2.984054e+06              Tucumán  129717.0        83.0   

      Fibra óptica  Wireless     Otros      Total  
0        1799013.0  150290.0  101294.0  5044557.0  
1         203756.0    4493.0   48879.0  1504711.0  
2          56542.0    1384.0      81.0    74933.0  
3          64463.0    8404.0   12404.0   155543.0  
4          26356.0   30867.0    9930.0   170676.0  
...            ...       ...       ...        ...  
22459        381.0    4092.0    3264.0    26764.0  
22460       3059.0    5951.0     644.0   506612.

In [ ]:
#Renombramos para un mejor entendimiento.
df4_nac = df4_nac.rename(columns = {'OTROS': 'Otros Mbps'})
df4_nac = df4_nac.rename(columns = {'Otros': 'Otras conexiones'})
df4_nac = df4_nac.rename(columns = {'Total': 'Total Mbps-conexiones'})
df4_nac = df4_nac.rename(columns = {'Entre 30 Mbps': 'Mas de 30 Mbps'})
df4_nac = df4_nac.rename(columns = {'Provincia_y': 'Provincia'})

In [ ]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 23 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Mbps (Media de bajada)     22464 non-null  float64
 5   Trimestre.1                22464 non-null  object 
 6   Provincia_x                22464 non-null  object 
 7   Hasta 512 kbps             22464 non-null  object 
 8   Entre 512 Kbps y 1 Mbps    22464 non-null  float64
 9   Entre 1 Mbps y 6 Mbps      22464 non-null  float64
 10  Entre 6 Mbps y 10 Mbps     22464 non-null  float64
 11  Entre 10 Mbps y 20 Mbps    22464 non-null  float64
 12  Entre 20 Mbps y 30 Mbps    22464 non-null  float64
 13  Mas de 30 Mbps             22464 non-null  flo

In [ ]:
columns_to_convert = [
    'Ingresos (miles de pesos)', 
    'ADSL', 
    'Cablemodem', 
    'Fibra óptica', 
    'Wireless', 
    'Otras conexiones', 
    'Total Mbps-conexiones'
]

df4_nac[columns_to_convert] = df4_nac[columns_to_convert].astype(int)

In [ ]:
# cambio los'.' por ','
df4_nac['Hasta 512 kbps'] = df4_nac['Hasta 512 kbps'].str.replace(',', '.')

In [ ]:
print(df4_nac['Entre 512 Kbps y 1 Mbps'].dtype)


float64


In [ ]:
df4_nac['Entre 512 Kbps y 1 Mbps'] = df4_nac['Entre 512 Kbps y 1 Mbps'].astype(str)


In [ ]:
df4_nac['Entre 512 Kbps y 1 Mbps'] = df4_nac['Entre 512 Kbps y 1 Mbps'].str.replace(',', '.')

In [ ]:
a_float = ['Hasta 512 kbps',
    'Entre 512 Kbps y 1 Mbps']
df4_nac[a_float] = df4_nac[a_float].astype(float)

In [ ]:
df4_nac.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22464 entries, 0 to 22463
Data columns (total 23 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        22464 non-null  int64  
 1   Trimestre                  22464 non-null  int64  
 2   Banda ancha fija           22464 non-null  int64  
 3   Dial up                    22464 non-null  int64  
 4   Mbps (Media de bajada)     22464 non-null  float64
 5   Trimestre.1                22464 non-null  object 
 6   Provincia_x                22464 non-null  object 
 7   Hasta 512 kbps             22464 non-null  float64
 8   Entre 512 Kbps y 1 Mbps    22464 non-null  float64
 9   Entre 1 Mbps y 6 Mbps      22464 non-null  float64
 10  Entre 6 Mbps y 10 Mbps     22464 non-null  float64
 11  Entre 10 Mbps y 20 Mbps    22464 non-null  float64
 12  Entre 20 Mbps y 30 Mbps    22464 non-null  float64
 13  Mas de 30 Mbps             22464 non-null  flo

-----

In [ ]:
df2_partidos.info(3)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2804 entries, 0 to 2803
Data columns (total 148 columns):
 #    Column          Dtype  
---   ------          -----  
 0    Provincia       object 
 1    Partido         object 
 2    Localidad       object 
 3    Link Indec_x    object 
 4    ADSL_x          int64  
 5    CABLEMODEM      int64  
 6    DIAL UP         int64  
 7    FIBRA OPTICA    int64  
 8    OTROS           int64  
 9    SATELITAL       int64  
 10   WIMAX           int64  
 11   WIRELESS        int64  
 12   Total general   int64  
 13   Población       int64  
 14   ADSL_y          object 
 15   Cablemódem      object 
 16   Dial Up         object 
 17   Fibra óptica    object 
 18   Satelital       object 
 19   Wireless        object 
 20   Telefonía Fija  object 
 21   3G              object 
 22   4G              object 
 23   Link            float64
 24   Latitud         float64
 25   Longitud        float64
 26   Link Indec_y    object 
 27   Otros           

In [ ]:
df2_partidos = df2_partidos.merge(mapa_conect[['Partido', 'Latitud', 'Longitud']], on='Partido', how='left')


In [ ]:
df2_partidos.sample(5)

Provincia             Partido         Localidad Link Indec_x  \
3984   BUENOS AIRES  General San Martín      Villa Yapeyú      6371010   
45273      SANTA FE    General Obligado     Arroyo Ceibal     82049010   
8113   BUENOS AIRES               Tigre          Nordelta      6805010   
19610       CORDOBA           San Justo  Altos de Chipión     14140020   
32614       MENDOZA          San Rafael        25 de Mayo     50105220   

       ADSL_x  CABLEMODEM  DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  ...  \
3984        0         265        0           540     13          0  ...   
45273       0           0        0             0      0          0  ...   
8113        0        5707        0          3831      0          0  ...   
19610       0           0        0           622      0          1  ...   
32614       0          79        1             0      0          0  ...   

       800 Mbps  850 Mbps  900 Mbps  999 Mbps 1000 Mbps 1024 Mbps 4000 Mbps  \
3984        NaN       NaN       NaN       NaN      42.0       NaN       NaN   
45273       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
8113        NaN       NaN       NaN       NaN     505.0       NaN       NaN   
19610       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
32614       NaN       NaN       NaN       NaN       NaN       NaN       NaN   

      10000 Mbps  Latitud_y Longitud_y  
3984         NaN -34.568213 -58.524545  
45273        NaN -28.230208 -59.263984  
8113         NaN -34.442567 -58.617436  
19610        NaN -30.383683 -60.330725  
32614        NaN -34.843315 -67.767420  

[5 rows x 150 columns]

In [ ]:
df2_partidos.dtypes

Provincia        object
Partido          object
Localidad        object
Link Indec_x     object
ADSL_x            int64
                 ...   
1024 Mbps       float64
4000 Mbps       float64
10000 Mbps      float64
Latitud_y       float64
Longitud_y      float64
Length: 150, dtype: object

In [ ]:
# Cambio de Dtypes
tipo = {'Provincia': str,
        'Partido': str,
        'Localidad': str}

df2_partidos = df2_partidos.astype(tipo)

In [ ]:
# Renombro Columnas
nuevo_nombre = {'ADSL_x': 'ADSL',
                'FIBRA OPTICA': 'Fibra Optica',
                'OTROS': 'Otras Conexiones',
                'CABLEMODEM': 'Cable Modem',
                'SATELITAL': 'Satelital',
                'Telefonía Fija': 'Telefonia Fija',
                'Latitud_y': 'Latitud',
                'Longitud_y': 'Longitud'}

In [ ]:
df2_partidos = df2_partidos.rename(columns = nuevo_nombre)

In [ ]:
#Visualizo
df2_partidos.head(5)

Provincia     Partido   Localidad Link Indec_x  ADSL  Cable Modem  \
0  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   
1  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   
2  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   
3  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   
4  BUENOS AIRES  25 de Mayo  25 de Mayo      6854100   755         4600   

   DIAL UP  Fibra Optica  Otras Conexiones  Satelital  ...  800 Mbps  \
0        0             2                 0        742  ...       NaN   
1        0             2                 0        742  ...       NaN   
2        0             2                 0        742  ...       NaN   
3        0             2                 0        742  ...       NaN   
4        0             2                 0        742  ...       NaN   

   850 Mbps  900 Mbps  999 Mbps 1000 Mbps 1024 Mbps 4000 Mbps 10000 Mbps  \
0       NaN       NaN       NaN       NaN       NaN       NaN        NaN   
1       NaN       NaN       NaN       NaN       NaN       NaN        NaN   
2       NaN       NaN       NaN       NaN       NaN       NaN        NaN   
3       NaN       NaN       NaN       NaN       NaN       NaN        NaN   
4       NaN       NaN       NaN       NaN       NaN       NaN        NaN   

     Latitud   Longitud  
0 -35.433939 -60.173121  
1 -35.739217 -60.559063  
2 -35.897345 -60.731609  
3 -35.270234 -59.557503  
4 -35.164397 -60.081358  

[5 rows x 150 columns]

----

In [ ]:
indic_me.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Unnamed: 0      22 non-null     int64  
 1   Año             22 non-null     int64  
 2   Trimestre       22 non-null     int64  
 3   IPC US          22 non-null     int64  
 4   IPC AR          22 non-null     int64  
 5   USDARS oficial  22 non-null     float64
 6   USDARS blue     22 non-null     float64
 7   PBI millones    22 non-null     int64  
dtypes: float64(2), int64(6)
memory usage: 1.5 KB


In [ ]:
# Elimino la columna  Unnamed: 0 
indic_me = indic_me.drop(columns=['Unnamed: 0'])

In [ ]:
indic_me

Año  Trimestre  IPC US  IPC AR  USDARS oficial  USDARS blue  PBI millones
0   2013          1  232773     298             5.0          9.0       2888981
1   2013          2  233504     309             5.0          9.0       3387811
2   2013          3  234149     331             6.0         10.0       3436547
3   2013          4  233049     351             6.0         11.0       3679896
4   2014          1  236293     402             8.0         11.0       3917649
5   2014          2  238343     433             8.0         13.0       4702630
6   2014          3  238031     461             8.0         14.0       4685503
7   2014          4  234812     491             9.0         13.0       5010564
8   2015          1  236119     518             9.0         13.0       5092694
9   2015          2  238638     549             9.0         15.0       5951479
10  2015          3  237945     584             9.0         15.0       6221731
11  2015          4  236525     621            10.0         15.0       6552140
12  2016          1  238132     697            14.0         15.0       6962845
13  2016          2  241038     751            14.0         15.0       8401125
14  2016          3  241428     791            15.0         16.0       8448890
15  2016          4  241432     833            15.0         16.0       8942135
16  2017          1  243801     880            16.0         16.0       9157377
17  2017          2  244955     925            16.0         18.0      10595155
18  2017          3  246819     963            17.0         18.0      10937240
19  2017          4  246524    1014            18.0         20.0      11533613
20  2018          1  249554    1083            20.0         24.0      11975804
21  2018          2  251989    1169            24.0         29.0             0

In [ ]:
#  
df4_comb.to_csv('df_combinado.csv',)
df_group.to_csv('df_agrupado.csv')
df4_nac.to_csv('df_nac.csv')
df2_partidos.to_csv('df_partido.csv')
indic_me.to_csv('df_indicadores.csv')                                                                                                                         